test/jtreg-ext/requires/VMProps.java

```diff
@@ -622,6 +683,23 @@ protected static void dump(Map<String, String> map) {
         }
     }
 
+    /**
+     * Logs diagnostic message.
+     *
+     * @param msg
+     */
+    protected static void log(String msg) {
+        if (!Boolean.getBoolean("jtreg.log.vmprops")) {
+            return;
```

Comment:
Why not always log in to the file like vmprops.log? Also, add try/catch into the call() and redirect file to stderr if something is wrong. And redirect if the property is set for passed tests.

Reply:
I had a discussion with Leonid in person, and I am OK with such approach. To summarize:
  - always log to a file (vmprops.log or similar)
  - this way we always have information to base investigation on if things go wrong
  - conditionally log into stderr to avoid polluting the output and being too verbose
If I do not hear any objections I will proceed with implementation tomorrow.